torch/csrc/jit/tensorexpr/external_functions.cpp

```diff
@@ -196,6 +204,29 @@ void nnc_aten_mean(
   }
 }
 
+void nnc_aten_addmm(
+    int64_t bufs_num,
+    void** buf_data,
+    int64_t* buf_ranks,
+    int64_t* buf_dims,
+    int8_t* buf_dtypes,
+    int64_t args_num,
+    int64_t* extra_args) {
```

Comment:
why not make this alfa and beta explicitly? is it just to avoid many declarations for now?

Reply:
> why not make this alfa and beta explicitly? is it just to avoid many declarations for now?

I think so - this way, the upstream caller side (IR Node -> external function call TensorExpr) can be consolidated into a simple shared helper function. @ZolotukhinM / @bertmaher might have more context.

It's a manual process to add new external functions today, and the overload resolution is ad-hoc, too (checkout how conv2d handles nullable bias param).

It's ok as we are only working with a small set of models for now. We should build a scalable way to do this - presumably via codegen from native_functions.yaml. The external_functions implementation can definitely be generated, maybe the IR node -> external_function part can be codegen-ed as well, then the signature here can be changed to align with aten op signature, which can reduce a little overhead of argument translation.

Reply:
Correct, all the wrappers must have the same signature, that's why we use generic "extra_args" here. This way it works for addmm, conv2d, average_pool_2d and presumingly all other ops. I'd expect it to stay this way even if we switch to automatic generation of these functions, because having a single signature for all wrappers makes llvm codegen much simpler.